### PR TITLE
Redesign: Library header

### LINKED ISSUE #6207: Redesign: Library
## Feature Description

Update library design based on [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=257%3A817).

- [ ] Update library tab icons using the design system (awaiting a decision on icon sizing)
- [ ] Update tab-to-tabpanel border opacity (`divider.secondary`)
- [ ] Update current tab indicator (24px `interactiveBg.brandNormal`, but extended to 38px wide on text tab)
- [ ] Update upload button using design system
- [ ] Update text tab animation according to new designs linked above and this small gif:
elementNav
- [ ] Update tooltip to new component where applicable

Do not update search input, this will happen in #6322

Do not remove or change the image type dropdown.

| Current design | New(ish) design |
|-|-|
| Screen Shot 2021-02-04 at 23 00 34 | Screen Shot 2021-02-04 at 23 00 26 |

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by miina**
## Context
Updates the Library header tabs and the media upload button.

## User-facing changes
- Tabs have updated icons
- Text tab animation behavior has changed slightly -- the hovered state stays on the icon that was last hovered on.
- Upload button is using the primary button from the design system,.

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Verify that the looks and behavior of the library tabs is matching [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3408%3A84442)
2. Verify the Media Upload button is matching [Figma](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=2241%3A2).

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6207

### CODE REVIEW

File: assets/src/design-system/theme/colors.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ const opacity = {
   white64: rgba(standard.white, 0.64),
   white24: rgba(standard.white, 0.24),
   white16: rgba(standard.white, 0.16),
-  white8: rgba(standard.white, 0.8),
+  white8: rgba(standard.white, 0.08),
```
Comment (barklund) [inline assets/src/design-system/theme/colors.js:106]: 🤦

File: assets/src/edit-story/components/tabview/index.js [hunk 2]
```diff
@@ -56,27 +57,29 @@ const Tab = styled.li.attrs(({ isActive }) => ({
   cursor: pointer;
   border: none;
   background: none;
-  color: ${({ theme }) => theme.DEPRECATED_THEME.colors.fg.white};
+  color: ${({ theme }) => theme.colors.fg.tertiary};
   font-family: ${({ theme }) => theme.DEPRECATED_THEME.fonts.tab.family};
   font-size: ${({ theme }) => theme.DEPRECATED_THEME.fonts.tab.size};
   font-weight: ${({ theme }) => theme.DEPRECATED_THEME.fonts.tab.weight};
   padding: 12px 0px;
```
Comment (barklund) [inline assets/src/edit-story/components/tabview/index.js]: This one should be `padding: 10px 0;`.

File: assets/src/edit-story/components/tabview/index.js [hunk 3]
```diff
@@ -56,27 +57,29 @@ const Tab = styled.li.attrs(({ isActive }) => ({
...
-  margin: 0px 16px;
+  margin: 0 22px;
   margin-bottom: -1px;
   position: relative;
 
   ${({ isActive, theme }) =>
     isActive &&
     `
-    border-bottom: 1px solid ${theme.DEPRECATED_THEME.colors.accent.primary};
+    border-bottom: 2px solid ${theme.colors.fg.primary};
```
Comment (barklund) [inline assets/src/edit-story/components/tabview/index.js:73]: And this one needs the padding-bottom reduced by the same 2px with: `padding-bottom: 8px`.

File: assets/src/edit-story/components/tabview/index.js [hunk 4]
```diff
@@ -56,27 +57,29 @@ const Tab = styled.li.attrs(({ isActive }) => ({
...
   border: none;
   background: none;
-  color: ${({ theme }) => theme.DEPRECATED_THEME.colors.fg.white};
+  color: ${({ theme }) => theme.colors.fg.tertiary};
   font-family: ${({ theme }) => theme.DEPRECATED_THEME.fonts.tab.family};
   font-size: ${({ theme }) => theme.DEPRECATED_THEME.fonts.tab.size};
   font-weight: ${({ theme }) => theme.DEPRECATED_THEME.fonts.tab.weight};
   padding: 12px 0px;
-  margin: 0px 16px;
+  margin: 0 22px;
```
Comment (barklund) [inline assets/src/edit-story/components/tabview/index.js]: This should be `margin: 0 20px;`, I think.

File: assets/src/edit-story/components/tabview/index.js [hunk 5]
```diff
@@ -39,12 +38,14 @@ const Tabs = styled.ul.attrs({
 })`
   display: flex;
   flex-direction: row;
+  align-items: stretch;
+  width: 100%;
+  max-width: 100%;
   justify-content: start;
```
Comment (barklund) [inline assets/src/edit-story/components/tabview/index.js]: And finally this one should be `justify-content: space-between;`.

**Review by barklund — COMMENTED** [review]
I played a bit around with the CSS to make the tabs look more identical to the Figma designs and have added some suggested changes. Bear in mind, that these changes (even just your changes) break the design for the inspector tabs (the alert icon is all the way out in the margin), as they now have too much spacing, but we can fix that later.

**Comment by miina** [comment]
> Bear in mind, that these changes (even just your changes) break the design for the inspector tabs (the alert icon is all the way out in the margin), as they now have too much spacing, but we can fix that later.

Yep, I saw that, it might make sense to just implement changes for the two panels together as well. WDYT?

**Comment by barklund** [comment]
> > Bear in mind, that these changes (even just your changes) break the design for the inspector tabs (the alert icon is all the way out in the margin), as they now have too much spacing, but we can fix that later.
> 
> Yep, I saw that, it might make sense to just implement changes for the two panels together as well. WDYT?

Yeah, feel free to tackle both tab views in the same PR. The other ticket is #6212, that would be solved then.

**Comment by miina** [comment]
> Yeah, feel free to tackle both tab views in the same PR. The other ticket is #6212, that would be solved then.

Hm, actually, it doesn't look _that_ horrible, the design panel tab, so I'll still leave it for a separate PR.

Applied your suggestions, thanks! Seems to match the measures from PR as far as I tested.

**Review by barklund — APPROVED** [review]

**Pull request merged**